src/Components/Forms/src/EditContextDataAnnotationsExtensions.cs

```diff
@@ -136,8 +155,125 @@ private void OnValidationRequested(object? sender, ValidationRequestedEventArgs
+        [UnconditionalSuppressMessage("Trimming", "IL2075", Justification = "Model types are expected to be defined in assemblies that do not get trimmed.")]
+        private static object GetFieldContainer(object obj, string fieldKey)
+        {
+            // The method does not check all possible null access and index bound errors as the path is constructed internally and assumed to be correct.
+            var dotSegments = fieldKey.Split('.')[..^1];
+            var currentObject = obj;
+
+            for (int i = 0; i < dotSegments.Length; i++)
+            {
+                string segment = dotSegments[i];
+
+                if (currentObject == null)
+                {
+                    string traversedPath = string.Join(".", dotSegments.Take(i));
+                    throw new ArgumentException($"Cannot access segment '{segment}' because the path '{traversedPath}' resolved to null.");
+                }
+
+                Match match = _pathSegmentRegex.Match(segment);
+                if (!match.Success)
+                {
+                    throw new ArgumentException($"Invalid path segment: '{segment}'.");
+                }
+
+                string propertyName = match.Groups[1].Value;
+                string? indexStr = match.Groups[2].Success ? match.Groups[2].Value : null;
+
+                Type currentType = currentObject.GetType();
+                PropertyInfo propertyInfo = currentType!.GetProperty(propertyName, BindingFlags.Public | BindingFlags.Instance)!;
+                object propertyValue = propertyInfo!.GetValue(currentObject)!;
+
+                if (indexStr == null) // Simple property access
+                {
+                    currentObject = propertyValue;
+                }
+                else // Indexed access
+                {
+                    if (!int.TryParse(indexStr, out int index))
+                    {
+                        throw new ArgumentException($"Invalid index '{indexStr}' in segment '{segment}'.");
+                    }
+
+                    if (propertyValue is Array array)
+                    {
+                        currentObject = array.GetValue(index)!;
+                    }
+                    else if (propertyValue is IList list)
+                    {
+                        currentObject = list[index]!;
+                    }
+                    else if (propertyValue is IEnumerable enumerable)
+                    {
+                        currentObject = enumerable.Cast<object>().ElementAt(index);
+                    }
+                    else
+                    {
+                        throw new ArgumentException($"Property '{propertyName}' is not an array, list, or enumerable. Cannot access by index in segment '{segment}'.");
+                    }
+                }
+
+            }
+            return currentObject!;
```

Comment:
For a prototype this is "ok", but we essentially have two options here. Either:
* We are able to get the container from the validation process as we walk the tree.
* We do something like what MVC does.
  * Process the string into a `System.Linq.Expression`, compile it and cache it (and I really don't want us to do this if we can avoid it).